gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -526,17 +531,40 @@ local function SetSubRoles(plys, availableRoles, selectableRoles)
 	end
 end
 
+---
+-- Ensure, that baseroles of hardforced subroles are accounted for (eg. received by ULX etc).
+--
+-- @return table List with a count of forced @{ROLE}s
+-- @realm server
+-- @internal
+local function GetHardForcedBaseRoles()
+	local selectedForcedRoles = {}
+
+	for ply, subrole in pairs(roleselection.finalRoles) do
+		if IsValid(ply) then
```

Comment:
That's not necessary

Reply:
What happens if you hardforce a player and then he leaves the game? Thought this would be safer or is it handled somewhere else?

Reply:
Yea that's something I forget, the list have to get cleaned up before the roleselection begins :)

Reply:
Okay, added a cleanup before forcing players.